includes/rest-api/class-amp-rest-fonts-controller.php

```diff
@@ -0,0 +1,242 @@
+		$response->header( 'X-WP-TotalPages', (int) $max_pages );
+
+		$request_params = $request->get_query_params();
+		$base           = add_query_arg( urlencode_deep( $request_params ), rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ) );
+
+		if ( $page > 1 ) {
+			$prev_page = $page - 1;
+
+			if ( $prev_page > $max_pages ) {
+				$prev_page = $max_pages;
+			}
+
+			$prev_link = add_query_arg( 'page', $prev_page, $base );
+			$response->link_header( 'prev', $prev_link );
+		}
+		if ( $max_pages > $page ) {
+			$next_page = $page + 1;
+			$next_link = add_query_arg( 'page', $next_page, $base );
+
+			$response->link_header( 'next', $next_link );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single font output for response.
+	 *
+	 * @param Object          $font Font object.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $font, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$schema = $this->get_item_schema();
+		$data   = [];
+
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $font['name'];
+		}
+
+		if ( in_array( 'slug', $fields, true ) ) {
+			$data['slug'] = $font['slug'];
+		}
+
+		if ( in_array( 'handle', $fields, true ) ) {
+			$data['handle'] = isset( $font['handle'] ) ? $font['handle'] : '';
+		}
+
+		if ( in_array( 'src', $fields, true ) ) {
+			$data['src'] = isset( $font['src'] ) ? $font['src'] : $schema['properties']['src']['default'];
+		}
+
+		if ( in_array( 'fallbacks', $fields, true ) ) {
+			$data['fallbacks'] = isset( $font['fallbacks'] ) ? (array) $font['fallbacks'] : $schema['properties']['fallbacks']['default'];
+		}
+
+		if ( in_array( 'weights', $fields, true ) ) {
+			$data['weights'] = isset( $font['weights'] ) ? (array) $font['weights'] : $schema['properties']['weights']['default'];
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filters a font returned from the API.
+		 *
+		 * Allows modification of the font right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param Object $font The original font object.
+		 * @param WP_REST_Request $request Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_font', $response, $font, $request );
+	}
+
+
+	/**
+	 * Checks if a given request has access to get fonts.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) {
+		return current_user_can( 'edit_posts' );
```

Comment:
Should this perhaps check the read access instead of edit?
The comment above mentions `read` access, too.

Reply:
The function returns if a user / request can read the fonts. The check if they can edit posts.

Reply:
This way the REST API endpoint is only accessible if you can edit posts, which makes sense to me.

Reply:
Yes, to think of it then it's only used in the admin anyway, the endpoint.